jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/EcsExplorerNodes.kt

```diff
@@ -0,0 +1,83 @@
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerResourceNode
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerServiceRootNode
+import software.aws.toolkits.jetbrains.core.explorer.nodes.ResourceParentNode
+import software.aws.toolkits.jetbrains.services.ecs.resources.EcsResources
+import software.aws.toolkits.resources.message
+
+class EcsParentNode(project: Project) : AwsExplorerServiceRootNode(project, AwsExplorerService.ECS) {
+    override fun getChildrenInternal(): List<AwsExplorerNode<*>> = listOf(
+        EcsClusterParentNode(nodeProject),
+        EcsTaskDefinitionFamiliesParentNode (nodeProject)
+    )
+}
+
+class EcsClusterParentNode(project: Project) :
+    AwsExplorerNode<String>(project, message("ecs.clusters"), AwsIcons.Logos.AWS),
+    ResourceParentNode {
+
+    override fun isAlwaysShowPlus(): Boolean = true
+
+    override fun getChildren(): List<AwsExplorerNode<*>> = super.getChildren()
+    override fun getChildrenInternal(): List<AwsExplorerNode<*>> = AwsResourceCache.getInstance(nodeProject)
+        .getResourceNow(EcsResources.LIST_CLUSTER_ARNS)
+        .map { EcsClusterNode(nodeProject, it) }
+}
+
+class EcsClusterNode(project: Project, private val clusterArn: String) :
+    AwsExplorerResourceNode<String>(project, EcsClient.SERVICE_NAME, clusterArn, AwsIcons.Logos.AWS),
+    ResourceParentNode {
+
+    override fun resourceType(): String = "cluster"
+    override fun resourceArn(): String = clusterArn
+    override fun displayName(): String = clusterArn.split("cluster/", limit = 2).last()
+    override fun isAlwaysShowPlus(): Boolean = true
+    override fun emptyChildrenNode(): AwsExplorerEmptyNode = AwsExplorerEmptyNode(nodeProject, message("ecs.no_services_in_cluster"))
+
+    override fun getChildren(): List<AwsExplorerNode<*>> = super<ResourceParentNode>.getChildren()
+    override fun getChildrenInternal(): List<AwsExplorerNode<*>> {
+        val resourceCache = AwsResourceCache.getInstance(nodeProject)
+        return AwsResourceCache.getInstance(nodeProject)
+            .getResourceNow(EcsResources.listServiceArns(clusterArn))
+            .map { resourceCache.getResourceNow(EcsResources.describeService(clusterArn, it)) }
+            .map { EcsServiceNode(nodeProject, it) }
+    }
+}
+
+class EcsServiceNode(project: Project, service: Service) :
+    AwsExplorerResourceNode<Service>(project, EcsClient.SERVICE_NAME, service, AwsIcons.Logos.AWS) {
+
+    override fun resourceType() = "service"
+    override fun resourceArn(): String = value.serviceArn()
+    override fun displayName(): String = value.serviceName()
+}
+
+class EcsTaskDefinitionFamiliesParentNode(project: Project) :
+    AwsExplorerNode<String>(project, message("ecs.task_definitions"), AwsIcons.Logos.AWS),
+    ResourceParentNode {
+
+    override fun isAlwaysShowPlus(): Boolean = true
+
+    override fun getChildren(): List<AwsExplorerNode<*>> = super.getChildren()
```

Comment:
Again isn't this just the default?

Reply:
Forces it to call the interface's default impl.